piperun/utils.py

```diff
@@ -12,7 +13,11 @@
 
 
 def parse_list(raw: dict, key: str, cast: Type[T]) -> list[T]:
-    return [cast(**value) for value in raw.get(key, [])]
+    raw_data = raw.get(key, [])
+    if not dataclasses.is_dataclass(cast):
+        return raw_data
+
+    return [cast(**value) for value in raw_data] # type: ignore
```

Comment:
Não retornar `raw_data`, Vamos fazer o cast para o tipo `cast` definido.

Eu havia pensado em algo parecido com isso:
```suggestion
    data = raw.get(key, [])
    if isinstance(data, dict):
        return [cast(**value) for value in data]
    else:
        return [cast(value) for value in data]
```